### ISSUE TITLE
Debian 12 - php 8.2.7

### ISSUE THREAD

**Issue by alexoalexo** [pnp4nagios/pnp4nagios#51]
Debian 12.4
Nagios® Core™ Version 4.4.6
PHP Version => 8.2.7

First error is about MB_OVERLOAD_STRING

`Fatal error: Uncaught Error: Undefined constant "MB_OVERLOAD_STRING" in /usr/local/pnp4nagios/lib/kohana/system/core/utf8.php:52 Stack trace: #0 /usr/local/pnp4nagios/lib/kohana/system/core/Bootstrap.php(32): require() #1 /usr/local/pnp4nagios/share/index.php(145): require('...') #2 {main} thrown in /usr/local/pnp4nagios/lib/kohana/system/core/utf8.php on line 52`

If removed from code I get the next one:

```
Please check the documentation for information about the following error.
Call to a member function getCode() on int
file [line]:
/usr/local/pnp4nagios/lib/kohana/system/core/Kohana.php [879]:
```

How to make it work on Debian 12.4?
Thanks

**Comment by sistemmsn** [comment]
> Debian 12.4 Nagios® Core™ Version 4.4.6 PHP Version => 8.2.7
> 
> First error is about MB_OVERLOAD_STRING
> 
> `Fatal error: Uncaught Error: Undefined constant "MB_OVERLOAD_STRING" in /usr/local/pnp4nagios/lib/kohana/system/core/utf8.php:52 Stack trace: #0 /usr/local/pnp4nagios/lib/kohana/system/core/Bootstrap.php(32): require() #1 /usr/local/pnp4nagios/share/index.php(145): require('...') #2 {main} thrown in /usr/local/pnp4nagios/lib/kohana/system/core/utf8.php on line 52`
> 
> If removed from code I get the next one:
> 
> ```
> Please check the documentation for information about the following error.
> Call to a member function getCode() on int
> file [line]:
> /usr/local/pnp4nagios/lib/kohana/system/core/Kohana.php [879]:
> ```
> 
> How to make it work on Debian 12.4? Thanks

Use php 8.1 in that version it works fine using the testing version k

**Comment by alexoalexo** [comment]
I've installed php 8.1, but not sure how to get/install  'version k'. 
Could you please elaborate more :) 
Thanks

**Comment by sistemmsn** [comment]
> Instalé php 8.1, pero no estoy seguro de cómo obtener/instalar la 'versión k'. ¿Podría dar más detalles :) Gracias

Of course, check issue #48  and you will find the details.

**Comment by alexoalexo** [comment]
I'm currently running php8.1 and pnp4nagios-0.6.27-4k.
My installation test page was all green, but now I don't see the graphs

What to check next?
Thanks
![1](https://github.com/pnp4nagios/pnp4nagios/assets/25418577/47968bf6-56dd-4dee-a7f6-9d317eeaed9d)

**Comment by alexoalexo** [comment]
The performance data is gathered and processed. I can see the xml/rrd files are updated.
And the perfdata.log currently has no errors.

**Comment by alexoalexo** [comment]
When trying the PDF export, I get :


```
Please check the documentation for information about the following error.

imagecreatefrompng(): &quot;/var/tmp/PNPFOhrVf&quot; is not a valid PNG file
file [line]:

application/models/rrdtool.php [290]:

[back](javascript:history.back())
```

I don't see anything created by pnp4nagios under /var/tmp,  could that be the issue of missing graphs?  Permissions or something else?

**Comment by celane** [comment]
I've been working (mostly offline) on a new release to handle source-code builds on RH-based and Debian-based systems.

Still trying on NetBSD, but progress is slow...just getting Apache running properly is a challenge :-( 

Most of the problems above are one's that I've seen/fixed with the released code (testing prerelease   version*-4l) 

Summary:   untar the source file
cd to source file 
autoconf
./configure
make all

If it's a NEW installation, decide whether you are using "sync", "bulk", or "npcd" to get data from Nagios to rrds
Sync doesn't work on some versions of Nagios, so maybe best to default to "bulk"

as root:
make use_bulk
make fulllinstall 

in /usr/local/pnp4nagios/etc   there's a 'verify-...'  script that checks some of the setup.
and when looking for a graph for the first time (before putting in a 'install.ignore' file) 
it also checks for config.

Usual problems on a new system are things like not having php modules for "mbstrings" "gd" "xml"  installed.
php -m lists the modules installed, so check.

**Comment by sistemmsn** [comment]
> I'm currently running php8.1 and pnp4nagios-0.6.27-4k. My installation test page was all green, but now I don't see the graphs
> 
> What to check next? Thanks ![1](https://private-user-images.githubusercontent.com/25418577/304397114-47968bf6-56dd-4dee-a7f6-9d317eeaed9d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDc4NDc5ODEsIm5iZiI6MTcwNzg0NzY4MSwicGF0aCI6Ii8yNTQxODU3Ny8zMDQzOTcxMTQtNDc5NjhiZjYtNTZkZC00ZGVlLWE3ZjYtOWQzMTdlZWFlZDlkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAyMTMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMjEzVDE4MDgwMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWJlZDQzZDk4NjkxNTkyMGU4YjRiYzFkZGI5MjgzYTE2N2UyMmQ4N2RjNjM5MWY2N2FmZWE3N2I2MDE5NDIyOWImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.o6YmwuwL4MM06dfna4pGiz-2J1l1tWwIovcqdjuG1dg)

It's strange to me that it pulls well, tested in almalinux 8.9

![image](https://github.com/pnp4nagios/pnp4nagios/assets/38023703/6853ed0e-910d-4be0-b41e-23f0aeb91ce9)

I edited the configure and the ac, in the problem that I mentioned there are my modifications, another one that I noticed that when I used make fullinstall in the end it threw an error for the nagios paths. So it is in this way:

using pnp4nagios-0.6.27-4k

cd /opt/pnp4nagios-0.6.27 && \ 
    ./configure && \
    make all && \ 
    make install && \
    make install-webconf && \
    make install-config && \
    make install-init && \
    make install-processperfdata && \
    make install-html && \
    make install-plugins

Another serious difference is that my entire infrastructure is dockerized 

![image](https://github.com/pnp4nagios/pnp4nagios/assets/38023703/840f88c4-b06e-41b0-a05b-f5618ad0a907)

![image](https://github.com/pnp4nagios/pnp4nagios/assets/38023703/b226e73a-8e3f-44af-9d03-ffdfcb46ee6e)

**Comment by alexoalexo** [comment]
After I deploy version k, I get
![1](https://github.com/pnp4nagios/pnp4nagios/assets/25418577/32fe08c2-6119-411b-9f2c-2d678b618881)


Is there a way to to turn some debug on,  so that we can find out why I don't get the graphs?

**Comment by celane** [comment]
Look at the bottom of that "status page", in the green box.

**Comment by alexoalexo** [comment]
> Look at the bottom of that "status page", in the green box.

I know, I just wanted to share that tests are OK, but after I 'touch'  install.ignore or remove the install.php I get no graphs as shown above.

So wondering if I can turn some debug option to on and see what the scripts are doing....

**Comment by alexoalexo** [comment]
![image](https://github.com/pnp4nagios/pnp4nagios/assets/25418577/3b4bc017-2dc7-40d6-b848-58f59523bd1b)

**Comment by sistemmsn** [comment]
> ![imagen](https://private-user-images.githubusercontent.com/25418577/304765166-3b4bc017-2dc7-40d6-b848-58f59523bd1b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDc5MjEyODgsIm5iZiI6MTcwNzkyMDk4OCwicGF0aCI6Ii8yNTQxODU3Ny8zMDQ3NjUxNjYtM2I0YmMwMTctMmRjNy00MGQ2LWI4NDgtNThmNTk1MjNiZDFiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAyMTQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMjE0VDE0Mjk0OFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWM2MDIyNWIwNGQ2ODBjNDllYWQ1Y2I4OTA4NzZjYzYwMjk2ZGE5YzUxM2Y4NzhiNTZhNzgzNWIzYTgyMDFjYzImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.P4YuFO2Rl4MxDe0j3lR5JU6VK4Hkt4OBiEfcyZ1z9AA)

The problem is going to be that they should inspect the page and check for errors, oh better switch to almalinux for something everything works fine, oh with docker I already have a nagios image with that version of pnp4nagios docker pull sistemmsn/nagios:4.4.14 and also the gaerman

**Comment by sistemmsn** [comment]
> > Mire la parte inferior de esa "página de estado", en el cuadro verde.
> 
> Lo sé, solo quería compartir que las pruebas están bien, pero después de "tocar" install.ignore o eliminar install.php no obtengo gráficos como se muestra arriba.
> 
> Entonces me pregunto si puedo activar alguna opción de depuración y ver qué están haciendo los scripts...

Your problem is, in how you are compiling it, why npcd does have a log output when it is generating the performance data, your configuration will be incorrectly referenced, due to its behavior obtained

**Comment by celane** [comment]
This is in the "testing" branch, most recent files, sample-config/SetLogLevels.in

(configure does substitutions for where files are, but you can do that manually if you want)

It's a simple script for setting log levels in npcd, process_perfdata, and the kohana web code, because they get their log level set in three different files.

**Comment by alexoalexo** [comment]
Thanks, I'll check it. I'm interested in kohana only for now. As I see that data is gathered and processed, but not visualized

**Comment by alexoalexo** [comment]
I have finally fixed it :)
![image](https://github.com/pnp4nagios/pnp4nagios/assets/25418577/97706037-92d6-4e25-88ea-e8cd6d942e1a)

The logging levels didn't help, but when looking in kohana config.php, I've noticed that $config['fontconfig_cache'] was pointing to a wrong apache cache dir.

After changing it to the correct one and changing ownership to www-data(debian), it worked!

Thanks!

**Comment by sistemmsn** [comment]
> I have finally fixed it :) ![image](https://private-user-images.githubusercontent.com/25418577/304825034-97706037-92d6-4e25-88ea-e8cd6d942e1a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDc5MzI2OTcsIm5iZiI6MTcwNzkzMjM5NywicGF0aCI6Ii8yNTQxODU3Ny8zMDQ4MjUwMzQtOTc3MDYwMzctOTJkNi00ZTI1LTg4ZWEtZThjZDZkOTQyZTFhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAyMTQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMjE0VDE3Mzk1N1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWM4MGYwMzA3MDEzYTU1ODUyMDk4OTJmMzM0YzUzMzA1ODc2NTE4Zjc5MmM0ZTJhOGQyMzYyMGIwOTRlYjk1M2EmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.igvCAj643q6Qb_wUgKNGmCI-8i0nfMct7WfjK8C6MtA)
> 
> The logging levels didn't help, but when looking in kohana config.php, I've noticed that $config['fontconfig_cache'] was pointing to a wrong apache cache dir.
> 
> After changing it to the correct one and changing ownership to www-data(debian), it worked!
> 
> Thanks!

It is for the same reason that he configures the references incorrectly.

**Comment by sistemmsn** [comment]
> config.php

Hello, how are you, can you explain how you solved it, I see 3 config.php, but what did you modify?

**Comment by sistemmsn** [comment]
> > config.php
> 
> Hello, how are you, can you explain how you solved it, I see 3 config.php, but what did you modify?

I saw it already

**Comment by sistemmsn** [comment]
I have tested in Debian 11.9 and almalinux 8.9 obtaining the same result

**Comment by celane** [comment]
Last iteration in the "testing" branch, tested to install/work on debian (and NetBSD and Alma) includes some changes that should deal with the problems mentioned in this issue. Since the testing branch has now been merged into the main branch and a new release 0.6.27-5 created, closing this issue. Please start a new issue if there are problems.
